code/_onclick/hud/fullscreen.dm

```diff
@@ -227,6 +261,248 @@
+		return
+	if(!is_ground_level(lighting_mob.z) && special_call != "z_change") // dont animate if not groundlevel
+		return
+
+	if(special_stage_time)
+		stage_time = special_stage_time
+
+	animate(screen, color = lighting_color, time = stage_time)
+
+
+/mob/proc/special_lighting_sunset(lighting_stage)
+
+	var/lighting_color = "#000" /// used in the animation, set by the special_lighting_type
+
+	//if its cold or in space (or chances claim)
+	if(SSmapping.configs[GROUND_MAP].environment_traits[MAP_COLD || ZTRAIT_IN_SPACE] || SSmapping.configs[GROUND_MAP].map_name == MAP_LV522_CHANCES_CLAIM)
+		switch(lighting_stage) //for sun stages, the more you have the better it looks when special called, I recommend choosing cinematic 4 colors then using a gradient tool to pick out the rest
+			if(0)
+				lighting_color = "#a8c3cf"
+			if(1)
+				lighting_color = "#7a9abb"
+			if(2)
+				lighting_color = "#6679a8"
+			if(3)
+				lighting_color = "#516a8b"
+			if(4)
+				lighting_color = "#38486e"
+			if(5)
+				lighting_color = "#2c2f4d"
+			if(6)
+				lighting_color = "#211b36"
+			if(7)
+				lighting_color = "#1f1b33"
+			if(8)
+				lighting_color = "#0c0a1b"
+			if(9)
+				lighting_color = "#000"
+	else //the default, a very warm sunset
+		switch(lighting_stage)
+			if(0)
+				lighting_color = "#e3a979"
+			if(1)
+				lighting_color = "#e29658"
+			if(2)
+				lighting_color = "#da8b4a"
+			if(3)
+				lighting_color = "#a9633c"
+			if(4)
+				lighting_color = "#90422d"
+			if(5)
+				lighting_color = "#68333a"
+			if(6)
+				lighting_color = "#4d2b35"
+			if(7)
+				lighting_color = "#231935"
+			if(8)
+				lighting_color = "#050c27"
+			if(9)
+				lighting_color = "#000"
+
+	return lighting_color
+
+
+/mob/proc/special_lighting_sunrise(lighting_stage)
+
+	var/lighting_color = "#000" /// used in the animation, set by the special_lighting_type
+
+	switch(lighting_stage) //only one type of sunrise cause it just looks more cinematic to have a warm sunset on cold maps
+		if(0)
+			lighting_color = "#000"
+		if(1)
+			lighting_color = "#040712"
+		if(2)
+			lighting_color = "#111322"
+		if(3)
+			lighting_color = "#291642"
+		if(4)
+			lighting_color = "#3f2239"
+		if(5)
+			lighting_color = "#632c3d"
+		if(6)
+			lighting_color = "#b97034" //it ends on very orange for cinematics
+
+	return lighting_color
+
+
+/mob/proc/special_lighting_register_signals()
+
+	var/mob/signaling_mob = src
+
+	RegisterSignal(signaling_mob, COMSIG_MOVABLE_Z_CHANGED, PROC_REF(special_lighting_z_change), TRUE)
+	RegisterSignal(signaling_mob, COMSIG_MOVABLE_ENTERED_AREA, PROC_REF(special_lighting_area_change), TRUE)
+
+
+/mob/proc/special_lighting_unregister_signals()
+
+	var/mob/signaling_mob = src
+
+	signaling_mob.special_lighting = null //clears special lighting
+
+	UnregisterSignal(signaling_mob, COMSIG_MOVABLE_Z_CHANGED, PROC_REF(special_lighting_z_change))
+	UnregisterSignal(signaling_mob, COMSIG_MOVABLE_ENTERED_AREA, PROC_REF(special_lighting_area_change))
```

Comment:
var/mob/signaling_mob = src, again just not needed but the maints might not care so idk,

Reply:
(we do)